### ISSUE TITLE
Memory Leak - 17GB Committed

### ISSUE THREAD

**Issue by camAtGitHub** [Artemis-RGB/Artemis#799]
After about 4 days of Artemis running in systray it used 17GB of memory.

It has happened for a while now and only just recently found out that Artemis was the cause.
Version: `1.2023.0612.3`
Don't sure what I can do to help troubleshoot... I have attached pictures and profile I was running.
![Before](https://github.com/Artemis-RGB/Artemis/assets/2858315/142a5819-6d65-491e-9bad-30ce4823294e)
![After](https://github.com/Artemis-RGB/Artemis/assets/2858315/483db7c0-b200-4b48-b5c1-787cc8e29e93)

Note that while there is RAM 'available' the committed RAM is so high programs begin erroring out and crashing...  

Cheers
[Fire - Blue creep - Green Caps.zip](https://github.com/Artemis-RGB/Artemis/files/11799019/Fire.-.Blue.creep.-.Green.Caps.zip)

**Comment by RobertBeekman** [comment]
Hi there, sorry for the inconvenience! Could you take a memory profile for us?
There's a guide here: https://wiki.artemis-rgb.com/en/guides/user/plugins/profiling

**Comment by camAtGitHub** [comment]
Before reading your message I completed a MemoryLeak Analysis using `DebugDiag` - Ill attach the report anyway... 

[Artemis.UI.Windows.exe__PID__9880_Manual Dump_MemoryAnalysis.mht.zip](https://github.com/Artemis-RGB/Artemis/files/11810603/Artemis.UI.Windows.exe__PID__9880_Manual.Dump_MemoryAnalysis.mht.zip)

**Comment by camAtGitHub** [comment]
JetBrains dump attached
[Artemis.UI.Windows.2023-06-21T10-30-58.974.dmw.zip](https://github.com/Artemis-RGB/Artemis/files/11810751/Artemis.UI.Windows.2023-06-21T10-30-58.974.dmw.zip)

**Comment by camAtGitHub** [comment]
And another for comparison. This one is after a restart of Artemis with Low mem usage (so far).  
[Artemis.UI.Windows.2023-06-21T11-00-18.674_Restarted_LowMemUsage.dmw.zip](https://github.com/Artemis-RGB/Artemis/files/11811066/Artemis.UI.Windows.2023-06-21T11-00-18.674_Restarted_LowMemUsage.dmw.zip)

**Comment by camAtGitHub** [comment]
Hello, I was wondering if you have had a chance to look at this yet, it's still happening on the current release (962b13c6ed888bf505fb3c183d43129764b5f866). I have taken to scripting a restart of Artemis every 3 hours for stability reasons.
Is there anything I can do to help?
Many Thanks

**Comment by RobertBeekman** [comment]
Hi, I'm sorry for the long delay, this has gone under the radar.

I downloaded the memory profiles but they don't contain any snapshots unfortunately. So while I can see the high usage I cannot see where it's from.

Again apologies for the delay, if this issue is still occuring please profile again and make one or two snapshots :)

**Comment by camAtGitHub** [comment]
@RobertBeekman - [Here you go a new memory profile upload](https://www.cammckenzie.com/wp/Artemis.UI.Windows.2023-08-22T10-20-45.931.dmw.7z) with multiple snapshots taken. 
Currently running v.1.2023.0821.1.
Many Thanks!

**Comment by RobertBeekman** [comment]
Thank you, I've found the issue.

I'll have a fix for this in the next update. In the meanwhile if you turn off the keypress brush the issue should go away. After the next update it'll no longer be a problem.

**Comment by RobertBeekman** [comment]
Hi, I've since deployed the release that fixes this :)

**Comment by camAtGitHub** [comment]
Hi @RobertBeekman 

Thanks for your work on this unfortunately there seems to still be a leak on v1.2023.0827.2, it got to 11GB before I noticed (and had to restart).
[Nonetheless I have taken a another capture over a few hours](https://www.cammckenzie.com/music/Artemis.UI.Windows.2023-08-30T17-52-34.271.dmw.zip)

Are you able to have another look? Want me to open another issue?

Thanks Again.

**Comment by RobertBeekman** [comment]
Hi there, does the [latest version](https://artemis-rgb.com/releases/windows/f10f4ccb-65a0-4c1c-a828-6a01ced025a8) fix your issue?

We've made some changes to how we check for running processes and are hoping it'll solve the leak you're experiencing

**Comment by camAtGitHub** [comment]
Hi @RobertBeekman 

No it doesn't seem to be fixed yet, currently running version: v1.2023.0902.4
New file upload from current version is [here](https://www.cammckenzie.com/music/Artemis.UI.Windows.20230906.dmw.zip)

Thanks Again

**Comment by camAtGitHub** [comment]
No probs, is there an easy way to backup/export everything? a db file or registry keys etc?

**Comment by RobertBeekman** [comment]
Yes, you can find the databse at `"C:\ProgramData\Artemis\database.db"`
If you'd rather not post it publicly feel free to mail it to support@artemis-rgb.com

**Comment by RobertBeekman** [comment]
I ran Artemis for about an hour with _all_ your profiles enabled but I did not observe the memory leak :\
Are you using any non-default plugins? 

Other than plugins the only difference I can think of is that I'm not using Razer, so perhaps there is an issue in there, I'll look into that

**Comment by camAtGitHub** [comment]
Hi Robert,

I was just replying to say I don't use the Razer provider as I dont want it's SDK installed and am using OpenRGB provider with OpenRGB also running locally.
I was checking my enabled plugins and noticed that both Razer and a few others I thought I had previously disabled were enabled so that maybe the issue as you say? But of note what Razer software I do have is disabled via system services. Perhaps the OpenRGB provider also leaks?

PS These stats are taken from today/just a few minutes ago. 
PPS Ill disable the Razer plugin and report back in a day or two.

![services msc](https://github.com/Artemis-RGB/Artemis/assets/2858315/d988d696-10a7-4fa4-9a84-92a90ad760f6)
![Artemis memory](https://github.com/Artemis-RGB/Artemis/assets/2858315/68121459-9129-4880-bf3c-7cda791fa1ea)

**Comment by camAtGitHub** [comment]
Hello,
Had Artemis running for 48 hours now with very good (low) memory usage. I have the Razer plugin disabled (fyi, so I'm not sure if that's fixed yet). Either way I'm happy to call this resolved. I'll let you close this (or not) incase you want to keep it open until the Razer plugin is fixed.
Either way - Thanks!

![image](https://github.com/Artemis-RGB/Artemis/assets/2858315/06bf4784-c29e-4ad4-a3e3-59d897a5dd0e)

**Comment by BadKiko** [comment]
Memory leak on arch linux

![image](https://github.com/Artemis-RGB/Artemis/assets/53761278/e7b6d78d-a792-4c0d-8215-dd0769f69ed8)
![image](https://github.com/Artemis-RGB/Artemis/assets/53761278/aa3fc7d2-0afe-41ef-97cd-2e065af17a6a)

**Comment by Myp3a** [comment]
Hello! I can confirm the same for latest version. After a few days without restarting, memory gets filled up and system slows down to an unusable state. Restarting Artemis fixes it for another few days. I'm not using Razer plugin directly, but there is a plugin that uses Razer SDK.

OS: Windows 11 23H2 (22631)
Artemis version: 1.2024.0503.2
Installed plugins except for default: Chroma Reader (probably from [here](https://github.com/diogotr7/Artemis.Plugins))
Workshop layers: AutoChroma
LED devices: SteelSeries laptop keyboard

I'll try to disable everything except for bare reproducible minimum, and then come back with a memory profile and results.

**Comment by Myp3a** [comment]
Got the issue once again, reporting now.  
I have updated everything to the latest versions and disabled other device plugins. Nothing criminal in the Task Manager or other utilities.

Got the memory profile - [memory_profile_high.zip](https://github.com/user-attachments/files/16400058/memory_profile_high.zip)
Also some screenshots with memory usage
<details><summary>Task Manager RAM with Artemis running for a while</summary>

![w_artemis](https://github.com/user-attachments/assets/e8f08938-0bd8-4573-bb40-2c955734490d)
</details>
<details><summary>Task Manager RAM after closing Artemis</summary>

![wo_artemis](https://github.com/user-attachments/assets/94bdf3d5-5f60-41e0-a190-85a57e225a92)
</details>

<details><summary>RAMMap summary</summary>

![rammap_summary](https://github.com/user-attachments/assets/3118a45d-d0c3-423f-a85c-d4a8ce60f442)
</details>

<details><summary>Task Manager commit size sorting</summary>

![commit](https://github.com/user-attachments/assets/bd4382f8-fda2-4586-a8ce-53e04b35d322)
</details>

Once again, updated system details:
OS: Windows 11 23H2 (22631)
Artemis version: 1.2024.0722.6
Installed plugins except for default: Chroma Reader (from [here](https://github.com/diogotr7/Artemis.Plugins)), latest from Git
Workshop layers: AutoChroma
LED devices: SteelSeries laptop keyboard
